src/main/java/io/anserini/collection/FileSegment.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Anserini: An information retrieval toolkit built on Lucene
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.anserini.collection;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.BufferedReader;
+import java.io.Closeable;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Iterator;
+
+/**
+ * A file containing one more source documents to be indexed. A collection is comprised of one or
+ * more {@code FileSegment}s. Note that a {@code FileSegment} can have independent existence
+ * outside a collection, and in principle multiple collections might share the same
+ * {@code FileSegment} implementation.
+ */
+public abstract class FileSegment<T extends SourceDocument> implements Iterator<T>, Closeable {
```

Comment:
How about rename `FileSegment` to `BaseFileSegment`? So that we do not have `public class FileSegment extends io.anserini.collection.FileSegment` inside each collection which does not look good.